enhancements/sig-architecture/76-addon-install-strategy/README.md

```diff
@@ -0,0 +1,182 @@
+# Addon Install Strategy
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+This proposal is to propose an enhancement of the `ClusterManagementAddon` API, so user can configure which clusters the related
+`ManagedClusterAddon` should be enabled.
+
+## Motivation
+
+We are seeing requirements that user would like the addon to be automatically installed when
+a new cluster is registered. We provided a [installStrategy interface](https://github.com/open-cluster-management-io/addon-framework/blob/main/pkg/agent/inteface.go#L56)
+in `addon-framework`, so the addon developer can select to automatically enable the `ManagedClusterAddon` in all or selected clusters. 
+However, this mechanism is not user friendly: 
+- An addon user cannot easily change the install strategy. For example, the user may want to disable the addon in certain clusters by deleting `ManagedClusterAddon`, but the addon-manager would recreate it.
+- It also confuses the addon developers, since we do not have an API standard to define the install strategy.
+
+### Goals
+
+- Enhance `ClusterManagementAddon` API to allow user to configure addon install strategy
+- A centralized addon-install-controller on hub being responsible of creation/deletion of `ManagedClusterAddon`
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+The admin can configure the `ClusterManagementAddon` API to automatically install a `ManagedClusterAddon` in all clusters
+or selected clusters.
+
+#### Story 2
+
+The admin can select NOT to install a `ManagedClusterAddon` in one or several clusters.
+
+#### Story 3
+
+The addon developer should not care about which cluster the `ManagedClusterAddon` should be installed.
+
+## Design Details
+
+### ClusterManagementAddon API change
+
+```go
+type InstallStrategy struct {
+	// Type is the type of the install strategy, it can be:
+	// - Manual: no automatic install
+	// - ClusterLabelSelector: install to clusters with certain labels
+	// - Placements: install to clusters selected by placements.
+	// +kubebuilder:validation:Enum=Manual;Placements
+	// +kubebuilder:default:=Manual
+	Type string `json:"type"`
+
+	// ClusterLabelSelector is a label selector honored when install strategy type is
+	// ClusterLabelSelector
+	ClusterLabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+
+	// Placements is a list of placement references honored when install strategy type is
+	// Placements. All clusters selected by these placements will install the addon
+	Placements []PlacementRef `json:"placements,omitempty"`
```

Comment:
when type is Placements, if the PlacementRefs is empty, what does it mean? or we do not allow PlacementRefs is empty when type is Placements

Reply:
we do not allow such configuration.